runtime/queries/zig/highlights.scm

```diff
@@ -1,231 +1,294 @@
-[
-  (container_doc_comment)
-  (doc_comment)
-] @comment.documentation
+; Comments
 
-[
-  (line_comment)
-] @comment.line
+((comment) @comment.block.documentation
+  (#lua-match? @comment.block.documentation "^//!"))
 
-;; assume TitleCase is a type
-(
-  [
-    variable_type_function: (IDENTIFIER)
-    field_access: (IDENTIFIER)
-    parameter: (IDENTIFIER)
-  ] @type
-  (#match? @type "^[A-Z]([a-z]+[A-Za-z0-9]*)+$")
-)
-
-;; assume camelCase is a function
-(
-  [
-    variable_type_function: (IDENTIFIER)
-    field_access: (IDENTIFIER)
-    parameter: (IDENTIFIER)
-  ] @function
-  (#match? @function "^[a-z]+([A-Z][a-z0-9]*)+$")
-)
-
-;; assume all CAPS_1 is a constant
-(
-  [
-    variable_type_function: (IDENTIFIER)
-    field_access: (IDENTIFIER)
-  ] @constant
-  (#match? @constant "^[A-Z][A-Z_0-9]+$")
-)
+(comment) @comment.line
 
-;; _
-(
-  (IDENTIFIER) @variable.builtin
-  (#eq? @variable.builtin "_")
-)
+; Punctuation
 
-;; C Pointers [*c]T
-(PtrTypeStart "c" @variable.builtin)
+[
+  "["
+  "]"
+  "("
+  ")"
+  "{"
+  "}"
+] @punctuation.bracket
 
 [
-  variable: (IDENTIFIER)
-  variable_type_function: (IDENTIFIER)
-] @variable
+  ";"
+  "."
+  ","
+  ":"
+  "=>"
+  "->"
+] @punctuation.delimiter
 
-parameter: (IDENTIFIER) @variable.parameter
+(payload "|" @punctuation.bracket)
 
-[
-  field_member: (IDENTIFIER)
-  field_access: (IDENTIFIER)
-] @variable.other.member
+; Literals
 
-[
-  function_call: (IDENTIFIER)
-  function: (IDENTIFIER)
-] @function
+(character) @constant.character
 
-exception: "!" @keyword.control.exception
+([
+  (string)
+  (multiline_string)
+] @string
+  (#set! "priority" 95))
```

Comment:
We don't support the priority property so the `(#set! ..)` here can be removed

Reply:
```suggestion
```